lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -5260,7 +5262,6 @@ SwiftASTContext::GetTypeInfo(opaque_compiler_type_t type,
                        .GetTypeInfo(pointee_or_element_clang_type);
     break;
   case swift::TypeKind::BoundGenericEnum:
-    swift_flags |= eTypeIsGeneric | eTypeIsBound;
```

Comment:
I like the idea of adding the new `eTypeHasUnboundGeneric` flag, which can be used to decide if binding is required, but why is it necessary to remove setting `eTypeIsGeneric` and `eTypeIsBound` everywhere? Isn't that potentially information we're losing? For example, for bound generic enums and structs we get the exact same flags as regular enums and structs.

Reply:
My thoughts were twofold:
- we're almost running out of bits in the 32-bit type flags (we can fix this with a more clever encoding, but it's extra work)
- all of the current clients of GetTypeInfo were asking whether the type had an unbound generic
GetTypeInfo() is really useful for writing things like data formatters while only using the public SBAPI.

Do we have a use-case for querying whether the top-level type is a (bound or unbound) generic, where we wouldn't have to immediately switch() over the actual type kind and do something that only TypeSystemSwift can do?

Reply:
I see. If we're not using them anywhere else and can't have more flags I think that's ok then

Reply:
I'd also be perfectly fine with reintroducing them if there is a need later.